esphome/components/sharp_memory_lcd/sharp_memory_lcd.cpp

```diff
@@ -0,0 +1,139 @@
+#include "sharp_memory_lcd.h"
+#include "esphome/core/log.h"
+#include "esphome/core/application.h"
+#include "esphome/components/display/display_buffer.h"
+
+namespace esphome {
+namespace sharp_memory_lcd {
+
+static const char *const TAG = "sharp_memory_lcd";
+
+#define TOGGLE_VCOM \
+  do { \
+    _sharpmem_vcom = _sharpmem_vcom ? 0x00 : SHARPMEM_BIT_VCOM; \
+  } while (0);
+
+void SharpMemoryLCD::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up SharpMemoryLCD...");
+  this->dump_config();
+  this->spi_setup();
+  this->init_internal_(this->get_buffer_length_());
+  this->cs_->digital_write(false);
+
+  if (this->extmode_) {
+    this->extmode_->digital_write(false);
+  }
+  if (this->extcomin_) {
+    this->extcomin_->digital_write(false);
+  }
+  if (this->disp_) {
+    this->disp_->digital_write(false);
+  }
+
+  display_init_();
+}
+
+void HOT SharpMemoryLCD::write_display_data() {
+  ESP_LOGD(TAG, "Writing display data...");
+
+  uint16_t i, currentline;
+
+  this->enable();
+  // Send the write command, this display has active HIGH
+  this->cs_->digital_write(true);
```

Comment:
The cs_pin should be inverted in the yaml config if this is the case.
```suggestion
  this->enable();
```

```yaml
display:
  platform: sharp_memory_lcd
  cs_pin:
    number :XX
    inverted: true
```

Reply:
I have documented this in the documentation PR, thanks!